src/Middleware/WebSockets/src/WebSocketMiddleware.cs

```diff
@@ -165,6 +155,54 @@ public async Task<WebSocket> AcceptAsync(WebSocketAcceptContext acceptContext)
 
                 return WebSocket.CreateFromStream(opaqueTransport, isServer: true, subProtocol: subProtocol, keepAliveInterval: keepAliveInterval);
             }
+
+            public static bool CheckSupportedWebSocketRequest(string method, IHeaderDictionary requestHeaders)
+            {
+                if (!string.Equals("GET", method, StringComparison.OrdinalIgnoreCase))
```

Comment:
```suggestion
                if (!HttpMethods.IsGet(method))
```
https://github.com/dotnet/aspnetcore/blob/9ab99b219c3a867a31c733103160c88394b9f120/src/Http/Http.Abstractions/src/HttpMethods.cs#L89